packages/flutter_migrate/lib/src/commands/resolve_conflicts.dart

```diff
@@ -0,0 +1,400 @@
+      } else {
+        stagingDirectory =
+            project.directory.childDirectory(customStagingDirectoryPath);
+      }
+    }
+    if (!stagingDirectory.existsSync()) {
+      logger
+          .printStatus('No migration in progress. Start a new migration with:');
+      printCommandText('start', logger, standalone: !isSubcommand);
+      return const CommandResult(ExitStatus.fail);
+    }
+
+    final File manifestFile =
+        MigrateManifest.getManifestFileFromDirectory(stagingDirectory);
+    final MigrateManifest manifest = MigrateManifest.fromFile(manifestFile);
+
+    checkAndPrintMigrateStatus(manifest, stagingDirectory, logger: logger);
+
+    final List<String> conflictFiles =
+        manifest.remainingConflictFiles(stagingDirectory);
+
+    terminal.usesTerminalUi = true;
+
+    for (int i = 0; i < conflictFiles.length; i++) {
+      final String localPath = conflictFiles[i];
+      final File file = stagingDirectory.childFile(localPath);
+      final List<String> lines = file.readAsStringSync().split('\n');
+      // We write a newline in the output, this counteracts it.
+      if (lines.last == '') {
+        lines.removeLast();
+      }
+
+      // Find all conflicts
+      final List<Conflict> conflicts = findConflicts(lines, localPath);
+
+      // Prompt developer
+      final CommandResult? promptResult =
+          await promptDeveloperSelectAction(conflicts, lines, localPath);
+      if (promptResult != null) {
+        return promptResult;
+      }
+
+      final bool result =
+          await verifyAndCommit(conflicts, lines, file, localPath);
+      if (!result) {
+        i--;
+      }
+    }
+    return const CommandResult(ExitStatus.success);
+  }
+
+  /// Parses the lines of a file and extracts a list of Conflicts.
+  List<Conflict> findConflicts(List<String> lines, String localPath) {
+    // Find all conflicts
+    final List<Conflict> conflicts = <Conflict>[];
+    Conflict currentConflict = Conflict.empty();
+    for (int lineNumber = 0; lineNumber < lines.length; lineNumber++) {
+      final String line = lines[lineNumber];
+      if (line.contains(_conflictStartMarker)) {
+        currentConflict.startLine = lineNumber;
+      } else if (line.contains(_conflictDividerMarker)) {
+        currentConflict.dividerLine = lineNumber;
+      } else if (line.contains(_conflictEndMarker)) {
+        currentConflict.endLine = lineNumber;
+        if (!(currentConflict.startLine == null ||
+                currentConflict.dividerLine == null ||
+                currentConflict.startLine! < currentConflict.dividerLine!) ||
+            !(currentConflict.dividerLine == null ||
+                currentConflict.endLine == null ||
+                currentConflict.dividerLine! < currentConflict.endLine!)) {
+          throw StateError(
+              'Invalid merge conflict detected in $localPath: Improperly ordered conflict markers.');
+        }
```

Comment:
Consider making this a simple state machine instead of checking at the end; I think that would make the logic simpler. E.g., have states for `lookingForConflictStart`, `lookingForConflictDivider`, `lookingForConflictEnd`, and then throw any time you hit something that isn't the one you are looking for.

That would also make it easy to include the line number in this error message.